src/pages/index.js

```diff
@@ -82,87 +59,105 @@ const api = new Api({
+      selectors: {
+        name: constant.profileTitle,
+        subtitle: constant.profileSubtitle,
+        link: constant.profileAvatar,
+      },
     });
+    userInfo.setUserFields();
   })
   .catch((err) => {
-    console.error(err); // log the error to the console
+    console.log(err);
   });
 
-const section = new Section(
-  { items: api.getInitialCards, renderer: createCard },
-  cardsList
-);
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  // destructure the response
+  .then(([userData, cards]) => {
+    // set all the data
+    userInfo = new UserInfo({
+      userData,
+      selectors: {
+        name: constant.profileTitle,
+        subtitle: constant.profileSubtitle,
+        link: constant.profileAvatar,
+      },
+    });
+    userInfo.setUserFields();
+    section.renderItems(cards);
+  })
+  .catch((err) => {
+    // catch possible errors
+  });
 
 function handleOpenEditProfileForm() {
   editProfileFormValidator.disableButton();
-  api.getUserInfo().then((result) => {
-    // process the result
-    profileTitleInputField.value = result.name;
-    profileSubtitleInputField.value = result.about;
-  });
+  const values = userInfo.getUserInfo();
+  addProfilePopup.setInputValues(values);
   addProfilePopup.openModal();
 }
 
-function handleAddNewCardButton() {
-  addNewCardFormValidator.disableButton();
-  newCardPopup.openModal();
-}
-
 function handleProfileFormSubmit(inputValues) {
-  document.querySelector(settings.submitButtonSelector).textContent =
-    "Saving...";
+  constant.submitButton.textContent = "Saving...";
   api
     .editUserInfo(inputValues)
-    .then((data) => {
-      // process the result
-      const name = data.name;
-      const subtitle = data.about;
-      return { name, subtitle };
+    .then((userData) => {
+      userInfo.updateUserData(userData);
+      userInfo.setUserFields();
+      addProfilePopup.closeModal();
+    })
+    .catch((err) => {
+      console.log(err);
     })
     .finally(() => {
-      document.querySelector(settings.submitButtonSelector).textContent =
-        "Save";
+      constant.submitButton.textContent = "Save";
     });
-  userInfo.setUserInfo(inputValues);
-  addProfilePopup.closeModal();
 }
 
-function handleCardDeleteFunctionInIndexComponent(cardId) {
-  api.deleteCard(cardId).then(() => {});
+function handleAddNewCardButton() {
+  addNewCardFormValidator.disableButton();
+  newCardPopup.openModal();
+}
+
+function createCard(item) {
+  // create instance of Card class
+  const card = new Card(
+    item,
+    "#card-template",
+    onCardClick,
+    handleDeleteCardBinButton,
+    onLikeButtonToggle,
+    userInfo
+  );
+  //create a card by calling getCardElement method from Card class
+  const cardElement = card.getCardElement();
+  //return the card
+  return cardElement;
+}
+
+function onLikeButtonToggle(cardId, status, callbackFn) {
+  if (status === "like") {
+    api.likeACard(cardId).then((data) => {
+      callbackFn(data);
+    });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**